plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/HttpThriftMetastoreClientFactory.java

```diff
@@ -0,0 +1,154 @@
+            checkArgument(httpThriftContext.authenticationMode.isPresent(), "'hive.metastore.http.client.authentication.type' must be set while using http/https metastore URIs in 'hive.metastore.uri'");
+            SSLConnectionSocketFactory socketFactory;
+            try {
+                socketFactory = new SSLConnectionSocketFactory(SSLContext.getDefault(), new DefaultHostnameVerifier(null));
+            }
+            catch (NoSuchAlgorithmException e) {
+                throw new TTransportException(e);
+            }
+            Registry<ConnectionSocketFactory> registry = RegistryBuilder.<ConnectionSocketFactory>create()
+                    .register("https", socketFactory)
+                    .build();
+            httpClientBuilder.setConnectionManager(new BasicHttpClientConnectionManager(registry));
+            httpClientBuilder.addRequestInterceptorFirst((httpRequest, entityDetails, httpContext) -> {
+                httpRequest.addHeader(HttpHeaders.AUTHORIZATION, "Bearer " + httpThriftContext.token.get());
+            });
+        }
+        else {
+            checkArgument(httpThriftContext.token.isEmpty(), "'hive.metastore.http.client.bearer-token' must not be set while using http metastore URIs in 'hive.metastore.uri'");
+        }
+        httpClientBuilder.addRequestInterceptorFirst((httpRequest, entityDetails, httpContext) -> {
+            httpThriftContext.additionalHeaders().forEach(httpRequest::addHeader);
+        });
+        httpClientBuilder.setDefaultRequestConfig(RequestConfig.custom().setResponseTimeout(readTimeoutMillis, TimeUnit.MILLISECONDS).build());
+        return new THttpClient(uri.toString(), httpClientBuilder.build());
+    }
+
+    private static Optional<SSLContext> buildSslContext(
+            boolean tlsEnabled,
+            Optional<File> keyStorePath,
+            Optional<String> keyStorePassword,
+            Optional<File> trustStorePath,
+            Optional<String> trustStorePassword)
+    {
+        if (!tlsEnabled) {
+            return Optional.empty();
+        }
+
+        try {
+            return Optional.of(createSSLContext(keyStorePath, keyStorePassword, trustStorePath, trustStorePassword));
+        }
+        catch (GeneralSecurityException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private record ThriftHttpContext(Optional<ThriftHttpMetastoreConfig.AuthenticationMode> authenticationMode, Optional<String> token, Map<String, String> additionalHeaders)
+    {
+        private ThriftHttpContext
+        {
+            requireNonNull(additionalHeaders, "additionalHeaders is null");
+            additionalHeaders = ImmutableMap.copyOf(additionalHeaders);
+        }
+    }
+
+    @VisibleForTesting
+    public static ThriftHttpContext buildThriftHttpContext(ThriftHttpMetastoreConfig config)
+    {
+        return new ThriftHttpContext(config.getAuthenticationMode(), config.getHttpBearerToken(), config.getAdditionalHeaders());
+    }
```

Comment:
Remove this method.

Reply:
This is also being used in the testHttpThriftConnection

Reply:
No.

Reply:
removed this.